server/controllers/incident.js

```diff
@@ -1,214 +1,195 @@
-            });
-        }
-        else {
-            res.status(400);
-            res.json({
-                status: 400,
-                message: `Bad Request`,
-                sampleData: '{createdOn:Date, createdBy:integer, type:string, location:string,status:string, imageUrl:string, videoUrl:string, comment:string}'
-            });
-        }
+/**
+ * API method to Post an Incident
+ * @static
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @return {object} JSON representing success message
+ * @memberof IncidentController
+*/
+
+  static createIncident(req, res) {
+    if (parseInt(req.body.createdBy, 10) > 0) {
+      const {
+        location, type, status, imageUrl, videoUrl, comment
+      } = req.body;
+      const id = db.incident[db.incident.length - 1].id + 1;
+      const createdOn = new Date().toString();
+      const createdBy = parseInt(req.body.createdBy, 10);
+      const newIncident = {
+        id, createdOn, createdBy, type, location, status, imageUrl, videoUrl, comment
+      };
+
+      db.incident.push(newIncident);
+      res.status(201);
+      res.json({
+        status: 201,
+        message: `Created ${newIncident.type} record`,
+        data: newIncident
+      });
+    } else {
+      res.status(400);
+      res.json({
+        status: 400,
+        message: 'Bad Request',
+      });
     }
-    // createIncident ends
-
-
-    /**
-     * API Get method to get all Incident
-     * @static
-     * @param {object} req - The request object
-     * @param {object} res - The response object
-     * @returns {object} JSON object representing success message
-     * @memberof IncidentController
-     */
-
-    static getAllIncident(req, res) {
-        if (db.incident.length !== 0) {
-            const type = req.url.split('/')[1];
-            const type2 = type.split('s')[0];
-            const data2 = db.incident.filter(incidentType => {
-                if (incidentType.type === type2) {
-                    return incidentType;
-                }
-            })
-
-            res.status(200);
-            res.json({
-                success: true,
-                message: `Successfully Retrieved all ${type2} `,
-                data: data2
-
-            });
-        }
-        else {
-            res.status(401);
-            res.json({
-                status: 401,
-                message: `You are not authorized to create Incident`,
-            });
+  }
+
+  /**
+  * API Get method to get all Incident
+  * @static
+  * @param {object} req - The request object
+  * @param {object} res - The response object
+  * @returns {object} JSON object representing success message
+  * @memberof IncidentController
+  */
+
+  static getAllIncident(req, res) {
+    if (db.incident.length !== 0) {
+      const type = req.url.split('/')[1];
+      const type2 = type.split('s')[0];
+      const data = db.incident.filter((incidentType) => {
```

Comment:
Make the variables used in this method more descriptive.

Reply:
Acknowledged and fixed